plugin/lighthouse/handler.go

```diff
@@ -14,38 +14,43 @@ import (
 // ServeDNS implements the plugin.Handler interface.
 func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
 	state := request.Request{W: w, Req: r}
+	qname := state.QName()
 
-	log.Debugf("Request received for  %q ", state.QName())
+	log.Debugf("Request received for %q", qname)
 
-	qname := state.QName()
 	// qname: mysvc.default.svc.example.org.
 	// zone:  example.org.
 	// Matches will return zone in all lower cases
 	zone := plugin.Zones(lh.Zones).Matches(qname)
 	if zone == "" {
-		return plugin.NextOrFailure(lh.Name(), lh.Next, ctx, w, r)
+		log.Debugf("Request does not match configured zones %v", lh.Zones)
+		return lh.nextOrFailure(state.Name(), ctx, w, r, dns.RcodeNotZone, "No matching zone found")
```

Comment:
@vthapar Note I changed this to call `lh.nextOrFailure` method to use the fall-through as in the other code paths below. Is this OK or was there a particular reason you called  `plugin.NextOrFailure` directly here?

Reply:
Just missed this in refactoring.